deepchecks/nlp/dataset.py

```diff
@@ -0,0 +1,557 @@
+    #                      stratify: t.Union[t.List, np.ndarray, bool] = False
+    #                      ) -> t.Tuple[TDataset, TDataset]:
+    #     """Split dataset into random train and test datasets.
+    #
+    #     Parameters
+    #     ----------
+    #     train_size : t.Union[int, float, None] , default: None
+    #         If float, should be between 0.0 and 1.0 and represent the proportion of the dataset to include in
+    #         the train split. If int, represents the absolute number of train samples. If None, the value is
+    #         automatically set to the complement of the test size.
+    #     test_size : t.Union[int, float] , default: 0.25
+    #         If float, should be between 0.0 and 1.0 and represent the proportion of the dataset to include in the
+    #         test split. If int, represents the absolute number of test samples.
+    #     random_state : int , default: 42
+    #         The random state to use for shuffling.
+    #     shuffle : bool , default: True
+    #         Whether to shuffle the data before splitting.
+    #     stratify : t.Union[t.List, np.ndarray, bool] , default: False
+    #         If True, data is split in a stratified fashion, using the class labels. If array-like, data is split in
+    #         a stratified fashion, using this as class labels.
+    #     Returns
+    #     -------
+    #     NLPData
+    #         Dataset containing train split data.
+    #     NLPData
+    #         Dataset containing test split data.
+    #     """
+    #     if isinstance(stratify, bool):
+    #         stratify = self.label_col if stratify else None
+    #
+    #     train_df, test_df = train_test_split(self._data,
+    #                                          test_size=test_size,
+    #                                          train_size=train_size,
+    #                                          random_state=random_state,
+    #                                          shuffle=shuffle,
+    #                                          stratify=stratify)
+    #     return self.copy(train_df), self.copy(test_df)
+
+    @property
+    def text(self) -> t.Sequence[str]:
+        """Return sequence of raw text samples.
+
+        Returns
+        -------
+        t.Sequence[str]
+           Sequence of raw text samples.
+        """
+        return self._dataset['text']
+
+    @property
+    def label_name(self) -> str:
+        """Return name of label.
+
+        Returns
+        -------
+        str
+            Name of label.
+        """
+        return self._label_name
+
+    @property
+    def label(self) -> TNLPLabel:
+        """Return the label defined in the dataset.
+
+        Returns
+        -------
+        TNLPLabel
+        """
+        return self._dataset[self.label_name]
+
+    @property
+    def classes(self) -> t.Tuple[str, ...]:
+        """Return the classes from label column in list. if no label column defined, return empty list.
```

Comment:
label column? this return all observed classes?
Does it work in token classification?

Reply:
yes